Assets/MixedRealityToolkit/_Core/EventDatum/Input/SourcePositionEventData.cs

```diff
@@ -37,46 +44,57 @@ public class SourcePositionEventData : SourceStateEventData
         /// <inheritdoc />
         public SourcePositionEventData(EventSystem eventSystem) : base(eventSystem) { }
 
+        public void Initialize(IMixedRealityInputSource inputSource, IMixedRealityController controller, TrackingState trackingState)
+        {
+            Initialize(inputSource, controller);
+            TrackingState = trackingState;
+        }
+
         /// <summary>
         /// Populates the event with data.
         /// </summary>
         /// <param name="inputSource"></param>
-        public void Initialize(IMixedRealityInputSource inputSource, Vector2 position)
+        /// <param name="controller"></param>
+        /// <param name="position"></param>
+        public void Initialize(IMixedRealityInputSource inputSource, IMixedRealityController controller, Vector2 position)
         {
-            Initialize(inputSource);
+            Initialize(inputSource, controller);
             TwoDofPosition = position;
         }
 
         /// <summary>
         /// Populates the event with data.
         /// </summary>
         /// <param name="inputSource"></param>
+        /// <param name="controller"></param>
         /// <param name="position"></param>
-        public void Initialize(IMixedRealityInputSource inputSource, Vector3 position)
+        public void Initialize(IMixedRealityInputSource inputSource, IMixedRealityController controller, Vector3 position)
         {
-            Initialize(inputSource);
+            Initialize(inputSource, controller);
             ThreeDofPosition = position;
         }
 
         /// <summary>
         /// Populates the event with data.
         /// </summary>
         /// <param name="inputSource"></param>
+        /// <param name="controller"></param>
         /// <param name="rotation"></param>
-        public void Initialize(IMixedRealityInputSource inputSource, Quaternion rotation)
+        public void Initialize(IMixedRealityInputSource inputSource, IMixedRealityController controller, Quaternion rotation)
         {
-            Initialize(inputSource);
+            Initialize(inputSource, controller);
             ThreeDofRotation = rotation;
         }
 
         /// <summary>
         /// Populates the event with data.
         /// </summary>
         /// <param name="inputSource"></param>
+        /// <param name="controller"></param>
         /// <param name="position"></param>
-        public void Initialize(IMixedRealityInputSource inputSource, SixDof position)
+        public void Initialize(IMixedRealityInputSource inputSource, IMixedRealityController controller, SixDof position)
```

Comment:
I'd rather see SixDof/Pose be it's own event / data as it is more than just position.

Reply:
I disagree, I don't want users to have lots of interlaces to implement to get similar data.

Reply:
I see where you are renaming PositionEventData to PoseEventData. I like that!